src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -98,6 +93,37 @@ public static FunctionPointerMethodSymbol CreateFromSource(FunctionPointerTypeSy
                 typeBinder,
                 diagnostics,
                 suppressUseSiteDiagnostics);
+
+            // PROTOTYPE(func-ptr): Do this right. Just making compatible changes for now
+            static CallingConvention getCallingConvention(FunctionPointerCallingConventionSyntax? callingConventionSyntax, DiagnosticBag diagnostics)
+            {
+                switch (callingConventionSyntax)
+                {
+                    case null:
+                        return CallingConvention.Default;
+                    case { ManagedSpecifier: { RawKind: (int)SyntaxKind.ManagedKeyword } }:
+                        Debug.Assert(callingConventionSyntax.UnmanagedCallingConventionList is null || callingConventionSyntax.ContainsDiagnostics);
+                        return CallingConvention.Default;
+                    case { ManagedSpecifier: { RawKind: (int)SyntaxKind.UnmanagedKeyword }, UnmanagedCallingConventionList: { } specifiers } when isCallingConvention(specifiers, "Cdecl"):
+                        return CallingConvention.CDecl;
+                    case { ManagedSpecifier: { RawKind: (int)SyntaxKind.UnmanagedKeyword }, UnmanagedCallingConventionList: { } specifiers } when isCallingConvention(specifiers, "Thiscall"):
+                        return CallingConvention.ThisCall;
+                    case { ManagedSpecifier: { RawKind: (int)SyntaxKind.UnmanagedKeyword }, UnmanagedCallingConventionList: { } specifiers } when isCallingConvention(specifiers, "Fastcall"):
+                        return CallingConvention.FastCall;
+                    case { ManagedSpecifier: { RawKind: (int)SyntaxKind.UnmanagedKeyword }, UnmanagedCallingConventionList: { } specifiers } when isCallingConvention(specifiers, "Stdcall"):
+                        return CallingConvention.Standard;
+                    case { ManagedSpecifier: { RawKind: (int)SyntaxKind.UnmanagedKeyword }, UnmanagedCallingConventionList: null }:
+                        return CallingConvention.Unmanaged;
+
+                    default:
+                        // PROTOTYPE(func-ptr): Handle the possible null and actually construct the correct convention
+                        diagnostics.Add(ErrorCode.ERR_InvalidFunctionPointerCallingConvention, callingConventionSyntax.GetLocation(), callingConventionSyntax.UnmanagedCallingConventionList!.CallingConventions[0].Name.Text);
+                        return CallingConvention.Default;
+                }
+
+                static bool isCallingConvention(FunctionPointerUnmanagedCallingConventionListSyntax specifiers, string expected)
+                    => specifiers.CallingConventions[0].Name.Text == expected;
```

Comment:
>specifiers.CallingConventions[0].Name.Text == expected [](start = 23, length = 54)

For now, we probably also want to report an error when there is more than one identifier in the list.